src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/model/EphemeronObject.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2017-2025 Software Architecture Group, Hasso Plattner Institute
+ * Copyright (c) 2021-2025 Oracle and/or its affiliates
+ *
+ * Licensed under the MIT License.
+ */
+package de.hpi.swa.trufflesqueak.model;
+
+import java.lang.ref.ReferenceQueue;
+import java.lang.ref.WeakReference;
```

Comment:
It seems `ReferenceQueue` and  `WeakReference` are not used yet. Is this intentional?

Reply:
This was accidentally left in because I had initially started from WeakVariablePointersObject.

They will most likely be needed for the final implementation though.

Reply:
I was wondering how you would want to do this, because ephemerons aren't a thing in Java. But, as you pointed out, it seems we can emulate them the same way it's done in [GraalJS's WeakMap](https://github.com/oracle/graaljs/blob/master/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/util/WeakMap.java#L59).

Do you want to do this as part of this PR or in a follow up?